sno/working_copy/gpkg.py

```diff
@@ -646,3 +666,21 @@ def _update_gpkg_contents(self, sess, dataset, commit=None):
                 {"last_change": gpkg_change_time, "table_name": dataset.table_name},
             ).rowcount
         assert rc == 1, f"gpkg_contents update: expected 1Δ, got {rc}"
+
+
+class GeometryType(UserDefinedType):
+    """UserDefinedType so that GPKG geometry is normalised to V2 format."""
+
+    def result_processor(self, dialect, coltype):
+        def process(gpkg_bytes):
```

Comment:
is this called lazily on column access? Could it be? Parsing and validating geometries is expensive.

Reply:
I don't know - but, there is no real benefit to doing so at present:
- The only reason we need to see what is actually in the user tables is to generate a diff
- When generating git-tree <> working-copy diffs, we need to check the entirety of each row (that is being tracked), to see what the diffs are
- We don't generate any of the diff lazily - we generate the whole diff before we output any of it. (For git-tree <> working-copy diffs). This could be rewritten so that it does work lazily, but with some difficulty - for instance, hard to see how we would keep rename-detection. The lazy-diff-output work I did before was entirely for git-tree <> git-tree diffs.

So, whether or not we make sure this is lazily-evaluated, the next step is for it to end up in basically a dict of all diffs that are then searched for renames. So not worth fixing except as part of a broader change to optimise git-tree <> working copy diffs.

Reply:
yup, that makes sense.